server/src/main/java/io/crate/auth/JWTAuthenticationMethod.java

```diff
@@ -80,18 +94,42 @@ public JWTAuthenticationMethod(Roles roles,
         }
 
         try {
-            JwkProvider jwkProvider = urlToJwkProvider.apply(decodedJWT.getIssuer());
-            Algorithm algorithm = resolveAlgorithm(jwkProvider, decodedJWT);
-
+            String issuer;
+            String name;
+            var defaultAud = jwtDefaults.get(AUD_KEY);
+            String audience = defaultAud != null ? defaultAud : clusterId.get();
             var jwtProperties = user.jwtProperties();
-            assert jwtProperties != null : "Resolved user must have jwt properties";
+            if (jwtProperties != null) {
+                issuer = jwtProperties.iss();
+                name = jwtProperties.username();
+                // Audience precedence: user defined -> default -> clusterId.
+                audience = jwtProperties.aud() != null ? jwtProperties.aud() : audience;
+            } else {
+                issuer = jwtDefaults.get(ISS_KEY);
+                // For users without JWT properties, token dictates CrateDB user.
+                // At this point username is resolved from claim, so withClaim("username", name)
+                // compares with "itself" as an extra check.
+                name = username;
+            }
+
+            if (issuer == null) {
+                issuer = "";
+                // Default is not set, set to empty string to fail on claim check.
+                // Saves an extra round of throw -> catch below -> throw again
+            }
+            assert name != null : "Must be defined on CREATE USER or equal to resolved by 'username' claim";
+            assert audience != null : "Must be defined on CREATE USER or taken from default or use cluster id";
```

Comment:
Also `audience` cannot be NULL at this point, all checks are already done above.
```suggestion
```